src/BloomExe/WebLibraryIntegration/BloomS3Client.cs

```diff
@@ -344,8 +344,13 @@ public string DownloadBook(string storageKeyOfBookFolder, string pathToDestinati
 			if (!GetBookExists(storageKeyOfBookFolder))
 				throw new DirectoryNotFoundException("The book we tried to download is no longer in the BloomLibrary");
 
+			// Amazon.S3 appears to truncate titles at 50 characters when building directory and filenames.  This means
+			// that relative paths can be as long as 117 characters (2 * 50 + 2 for slashes + 15 for .BloomBookOrder).
+			// So our temporary folder must be no more than 140 characters (allow some margin) since paths can be a
+			// maximum of 260 characters in Windows.  Including the storageKeyOfBookFolder in the path can push it over
+			// the limit.  See https://jira.sil.org/browse/BL-1160.
 			using (var tempDestination =
-					new TemporaryFolder("BloomDownloadStaging_" + storageKeyOfBookFolder + "_" + Guid.NewGuid()))
```

Comment:
I'd be inclined to shorten it further..maybe BDS_ if you want to group them in the temp folder. I'm not sure there's any limit on how long an image file name might be, so the more margin the better.